crates/shared/src/bad_token/trace_call.rs

```diff
@@ -166,35 +184,44 @@ impl TraceCallDetector {
             Some(amount) => amount,
             None => {
                 return Ok(TokenQuality::bad(
-                    "token total supply does not fit a uint256",
+                    "Transferring {amount} into settlement contract would overflow its balance.",
                 ))
             }
         };
         if balance_after_in != computed_balance_after_in {
             return Ok(TokenQuality::bad(
-                "balance after in transfer does not match",
+                "Transferring {amount} into settlement contract was expected to result in a \
+                 balance of {computer_balance_after_in} but actually resulted in \
+                 {balance_after_in}.",
```

Comment:
This error case is also pretty common. I think we could also give a best effort hint what might be the exact problem (like for the vyper bug). In this case we could mention that the token could take a fee on transfer. This doesn't explain every such error but probably the majority of them.

Reply:
added